pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -550,18 +550,55 @@ func (service *SubnetPortService) AllocatePortFromSubnet(subnet *model.VpcSubnet
 
 	info.lock.Lock()
 	defer info.lock.Unlock()
+
+	dhcpMode := "DHCP_DEACTIVATED"
+	subnetInfo, _ := servicecommon.ParseVPCResourcePath(*subnet.Path)
+	if subnet.SubnetDhcpConfig != nil && subnet.SubnetDhcpConfig.Mode != nil {
+		dhcpMode = *subnet.SubnetDhcpConfig.Mode
+	}
+	// For DHCP Server mode Subnet, get total IPs from DHCP IP Pool from NSX each time
+	// since user might update reservedIPRanges for the subnet and it impacts the DHCP Pool size
+	if dhcpMode == "DHCP_SERVER" {
+		dhcpServerStats, err := service.NSXClient.StatsClient.Get(subnetInfo.OrgID, subnetInfo.ProjectID, subnetInfo.VPCID, subnetInfo.ID, nil, nil, nil, nil, nil, nil, nil)
+		if err != nil {
+			log.Error(err, "Failed to get Subnet dhcp-server-config stats", "Subnet", *subnet.Path)
+			return false
+		}
+		info.totalIP = int(*dhcpServerStats.IpPoolStats[0].PoolSize)
+	}
+
 	if !ok {
-		var totalIP int
-		if subnet.Ipv4SubnetSize != nil {
-			totalIP = int(*subnet.Ipv4SubnetSize)
+		// For DHCP Deactivated mode Subnet, get total IPs from IP pool static-ipv4-default
+		if dhcpMode == "DHCP_DEACTIVATED" {
+			staticIpAllocationEnabled := false
+			info.totalIP = 0
```

Comment:
It seems for static ip allocation with false, we will get into.totalIP=0. Maybe we need to update the code here to return true for staticIpAllocation disable case?

https://github.com/vmware-tanzu/nsx-operator/blob/eef6d631b6b04cda99b246f994f67b5842274b7d/pkg/nsx/services/subnetport/subnetport.go#L610-L615